core-java-8/src/main/java/com/baeldung/datetime/AgeCalculator.java

```diff
@@ -0,0 +1,28 @@
+package com.baeldung.datetime;
+
+import java.time.Period;
+
+import org.joda.time.Years;
+
+public class AgeCalculator {
+
+    private AgeCalculator() {
+    }
+
+    public static int calculateAge(java.time.LocalDate birthDate, java.time.LocalDate currentDate) {
+        if (birthDate == null || currentDate == null) {
+            throw new IllegalArgumentException("Please enter a valid date!");
+        }
+        return Period.between(birthDate, currentDate)
+            .getYears();

```

Comment:
Continuations should be indented in 2 spaces, not 4. However, you could also safely put this on the line above, if you wanted.

Reply:
Provide separate null check for date of birth and current date